repo/gromacs/package.py

```diff
@@ -515,18 +544,14 @@ def cmake_args(self):
                 options.append("-DGMX_GPU:STRING=CUDA")
             elif "+opencl" in self.spec:
                 options.append("-DGMX_GPU:STRING=OpenCL")
-            elif "+rocm" in self.spec:
-                archs = self.spec.variants["amdgpu_target"].value
-                arch_str = ",".join(archs)
-                options.extend(
-                    [
-                        "-DGMX_GPU:STRING=SYCL",
-                        "-DGMX_SYCL_HIPSYCL=ON",
-                        f"-DHIPSYCL_TARGETS=hip:{arch_str}",
-                    ]
-                )
-            elif "+sycl" in self.spec:
+            elif "+sycl" in self.spec or "+rocm" in self.spec:
                 options.append("-DGMX_GPU:STRING=SYCL")
+                if "+rocm" in self.spec:
+                    options.append("-DGMX_SYCL_HIPSYCL:BOOL=ON")
+                    hipsycl_dir = os.path.join(self.spec["sycl"].prefix.lib, "cmake/hipSYCL/")
+                    options.append(f"-Dhipsycl_DIR:STRING={hipsycl_dir}")
+                    rocm_archs = ",".join(self.spec.variants["amdgpu_target"].value)
+                    options.append(f"-DHIPSYCL_TARGETS:STRING=hip:{rocm_archs}")
```

Comment:
Thanks for submitting this @rfhaque 

A couple overall thoughts:

* This is similar to `https://github.com/spack/spack/pull/43108` but appears to remove a few bits: did you find that using the exact PR caused problems for you?
* This stacks the changes in `https://github.com/spack/spack/pull/43108` on top of `https://github.com/spack/spack/pull/41767` - the latter PR (which was merged to Benchpark, but not Spack) was intended to be sufficient for building `gromacs%rocmcc@5.4.3+rocm` - did you have problems with that (which led you to `spack/pull/43108`)? If so, could you mention what compiler you used?